### ISSUE TITLE
Ship Detection using DL

### ISSUE THREAD

**Issue by Kshah002** [abhisheks008/DL-Simplified#634]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : Ship Detection from Aerial Images</br>
:red_circle: **Aim** : To detect whether there are any ships in the given picture or not.</br>
:red_circle: **Dataset** : https://www.kaggle.com/datasets/andrewmvd/ship-detection</br>
:red_circle: **Approach** : 

Dataset being very scarce the first task to apply data augmentation 

Using Transfer Learning approach by leveraging pretrained models - DenseNet121, ResNet50, Xception and EfficientNet B5 and comparing the results. Will try to implement all four models however can assure to use atleast 3 models mentioned

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- GitHub Profile Link : 
- Email ID :
- Participant ID (if applicable):
- Approach for this Project :
- What is your participant role? (Mention the Open Source program)

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by abhisheks008** [comment]
This project is already present in this repo, https://github.com/abhisheks008/DL-Simplified/tree/main/Ship%20Detection%20from%20Aerial%20Images

**Comment by Kshah002** [comment]
ya i was thinking to work on the same statement and try to get better results with different models as they have used yolo over there. If it is okay ?

**Comment by abhisheks008** [comment]
Then it's an enhancement of the existing project. You can add your contribution in the existing project folder, no need to create a separate project folder.

Can you share the models you are planning to implement for the enhancement of the existing project?
@Kshah002

**Comment by Kshah002** [comment]
yeah okay so if you could provide me with the link of the folder and also label with necessary gssoc tags, it would be great
Here are the models I am planning to implement - 
DenseNet - https://keras.io/api/applications/densenet/
Xception - https://keras.io/api/applications/xception/
EfficientNet - https://keras.io/api/applications/efficientnet/
ResNet - https://keras.io/api/applications/resnet/

**Comment by abhisheks008** [comment]
https://github.com/abhisheks008/DL-Simplified/tree/main/Ship%20Detection%20from%20Aerial%20Images

Enhance the model only. No need to remove the files, edit the jupyter notebook and push your codes. Once done update the README file along with your models.

Assigned @Kshah002

**Comment by Kshah002** [comment]
👍Thank you.

**Comment by adityasingh-0803** [comment]
i wanna work on this

**Comment by abhisheks008** [comment]
Please share your approach as per the issue template along with the required details.
@adityasingh-0803

**Comment by adityasingh-0803** [comment]
Data Augmentation: Since the dataset is limited, apply data augmentation techniques like rotation, flipping, zooming, and shifting to artificially increase the diversity of training data.

Model Training: Fine-tune each of the pretrained models on the augmented dataset. The models will be trained to classify whether a ship is present or not in the aerial images.

Model Evaluation: Compare the performance of the models based on accuracy, precision, recall, and F1-score to determine which model performs best for ship detection.

Model Selection: Choose the best-performing model based on evaluation metrics and further optimize it if necessary.

Model Deployment: Save the trained models for later use and further analysis.

**Comment by adityasingh-0803** [comment]
please tell

**Comment by abhisheks008** [comment]
Hi @adityasingh-0803 sorry for replying late. I'd like to know about the models you are planning to implement here for this project/issue. You need to implement at least 3-4 Deep Learning models for this project. Also please mention the open source event you are participating in.

**Comment by adityasingh-0803** [comment]
ok i will work on this

**Comment by abhisheks008** [comment]
> ok i will work on this

Before starting the issue related work, can you please mention the details highlighted in the previous comment?
